plugins/frontend/backstage-plugin-argo-cd/src/components/useAppDetails.ts

```diff
@@ -139,8 +139,8 @@ export const useAppDetails = ({
             return newItem;
           },
         );
-        const result = await Promise.all(getRevisionHistroyPromises);
-        return result;
+        return Promise.all(getRevisionHistroyPromises)
```

Comment:
Let's not move away from using `await` which should be the default approach to handling Promises.

Reply:
@Xantier `return await` is an anti-pattern outside of `try-catch`. The [ESLint rule](https://eslint.org/docs/latest/rules/no-return-await) that prevents it explains why. Also here's a [blog post](https://jakearchibald.com/2017/await-vs-return-vs-return-await/) that goes a bit more in depth

Reply:
Thanks for the blog post. An alternative approach to `then` should be used regardless.